src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -50,28 +52,92 @@ public AspNetCoreDiagnosticObserver(Tracer tracer)
 
         protected override string ListenerName => DiagnosticListenerName;
 
+#if NETCOREAPP
         protected override void OnNext(string eventName, object arg)
         {
-            switch (eventName)
+            var lastChar = eventName[^1];
+
+            if (lastChar == 't')
             {
-                case "Microsoft.AspNetCore.Hosting.HttpRequestIn.Start":
+                if (eventName.AsSpan().Slice(PrefixLength).SequenceEqual("Hosting.HttpRequestIn.Start"))
+                {
                     OnHostingHttpRequestInStart(arg);
-                    break;
+                }
+
+                return;
+            }
 
-                case "Microsoft.AspNetCore.Mvc.BeforeAction":
+            if (lastChar == 'n')
+            {
+                var suffix = eventName.AsSpan().Slice(PrefixLength);
+
+                if (suffix.SequenceEqual("Mvc.BeforeAction"))
+                {
                     OnMvcBeforeAction(arg);
```

Comment:
Should we just return after `OnMvcBeforeAction(arg)` so no more comparisons are done?

Reply:
You're right, I missed that when I moved the code from the benchmark